### PR TITLE
Support s3 bucket configuration types

### PR DESCRIPTION
**Pull request by erran-r7**
These changes are required to support setting the CorsConfiguration through convection.

---

cc @abegley-r7 @simonirwin-r7 @mmckeaveney-r7

### CODE REVIEW

File: lib/convection/model/template/resource/aws_s3_bucket.rb [hunk 1]
```diff
@@ -13,12 +13,18 @@ class S3Bucket < Resource
           type 'AWS::S3::Bucket'
           property :bucket_name, 'BucketName'
           property :access_control, 'AccessControl'
-          property :cors_configurationm, 'CorsConfiguration'
+          property :cors_configuration, 'CorsConfiguration'
```
Comment (erran-r7) [inline lib/convection/model/template/resource/aws_s3_bucket.rb]: **NOTE**: This fix should be pulled in via #135.

File: lib/convection/model/template/resource_property/aws_s3_cors_configuration_rule.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
+require_relative '../resource_property'
+
+module Convection
+  module Model
+    class Template
+      class ResourceProperty
+        # Represents an {http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-s3-bucket-cors-corsrule.html
+        # Amazon S3 Cors Configuration Rule}
+        class S3CorsConfiguration < ResourceProperty
```
Comment (erran-r7) [inline lib/convection/model/template/resource_property/aws_s3_cors_configuration_rule.rb]: 😏 small typo here.

**Comment by abegley-r7** [comment]
Looks good, only concern is impacting current stacks, spoke with @e with regards to current usage, looks like no other stack references/makes use of CORS.

**Comment by erran-r7** [comment]
> Looks good, only concern is impacting current stacks, spoke with @erran-r7 with regards to current usage, looks like no other stack references/makes use of CORS.

@abegley-r7 spoke to @jmanero-r7 and I'm going to address that in #135 with a deprecation warning for the bad property name.

**Comment by erran-r7** [comment]
> I'm going to address that in #135 with a deprecation warning for the bad property name.

Done! Merging after tests complete.

**Pull request merged**